CHANGELOG.rst

```diff
@@ -1,5 +1,28 @@
 .. _changelog:
 
+0.23.0
+------
+
+Renku ``0.23.0`` brings number of bug fixes, KG APIs performance improvements as well as support for CLI 1.11.1.
+
+**🐞 Bug Fixes**
+
+* **Knowledge Graph**: Performance improvements for Cross-Entities Search and Project Details APIs.
+* **Knowledge Graph**: Partial support for Workflow File metadata.
```

Comment:
Could you be more specific about what this means? What is and is not supported? 
If it would be too long to get into the details here in the Changelog, is there a PR that you could link to from here where the specifics are described?

Reply:
Unfortunately, it's not a simple one. Generally speaking, for the time being, we'll treat the `WorkflowFile` specific `Plan` and `CompositePlan` entities as their non-WorkflowFile versions. As they are more or less the same (the WorklfowFile ones contain some additional properties) there's not much of a problem on the KG side to see them that way. That gives us that all the most essential information related to the WorkflowFile feature will be present in the Triples Store and queryable by our APIs. Obviously, information such as the path to the workflow file yaml or a group of activities generated during a single workflow execution won't be there. This should not be a problem as our APIs don't return this data anyway.
I hope that gives a bit of light. if not I'm more than happy to elaborate more :)

Reply:
Does this release make the lineage view in the file browser work for files created via a workflow file?

Reply:
Yes, that should work.

Reply:
only one way to find out!

Reply:
I opened the PR deployment and tried to view lineage on my Academic Team demo, and it [didn't work](https://ci-renku-2914.dev.renku.ch/projects/laura.kinkead/academic-team-demo-sklearn/files/lineage/data/05_metrics/model_comparison.png). I also tried with the workflow demo I'd been testing with Mohammad before the metadata 2.0 release, and lineage [doesn't work there either](https://ci-renku-2914.dev.renku.ch/projects/laura.kinkead/workflow-file-demo/files/lineage/data/output/filtered.csv). It's not clear to me if there's a different way I should be testing it.

From the user's point of view, the lineage view not working for file-based workflows is a regression. I think we should be careful to only let that regression into a release if it's intentional- i.e. we're aware of the technical tradeoffs and decide to make a release anyway (and communicate the regression to users). 

@jachro Would you be willing to the bottom of this and get a sure answer?

Reply:
Sorry @lokijuhy I've got a bit tendency to miss notifications from GH :)

So the problem with your project and lineage is caused by problems during `renku migrate` we call for each commit we process. I'm guessing it might be because of some incompatibilities between the renku 1.11.1 we use for triples generation and the one you used for the project. The error we received from the CLI looks like this:
```
File \"/usr/bin/renku\", line 8, in <module>; 
sys.exit(cli()); 
File \"/usr/lib/python3.10/site-packages/click/core.py\", line 1130, in __call__; 
return self.main(*args, **kwargs); 
File \"/usr/lib/python3.10/site-packages/renku/ui/cli/exception_handler.py\", line 133, in main; 
return super().main(*args, **kwargs); 
File \"/usr/lib/python3.10/site-packages/renku/ui/cli/exception_handler.py\", line 92, in main; 
return super().main(*args, **kwargs); 
File \"/usr/lib/python3.10/site-packages/click/core.py\", line 1055, in main; 
rv = self.invoke(ctx); 
File \"/usr/lib/python3.10/site-packages/click/core.py\", line 1657, in invoke; 
return _process_result(sub_ctx.command.invoke(sub_ctx)); 
File \"/usr/lib/python3.10/site-packages/click/core.py\", line 1404, in invoke; 
return ctx.invoke(self.callback, **ctx.params); 
File \"/usr/lib/python3.10/site-packages/click/core.py\", line 760, in invoke; 
return __callback(*args, **kwargs); 
File \"/usr/lib/python3.10/site-packages/renku/ui/cli/migrate.py\", line 98, in migrate; 
status = check_project().build().execute().output; 
File \"/usr/lib/python3.10/site-packages/renku/command/command_builder/command.py\", line 223, in execute; 
hook(self, context, *args, **kwargs); 
File \"/usr/lib/python3.10/site-packages/renku/command/command_builder/database.py\", line 84, in _injection_pre_hook; 
minimum_renku_version = Version(self.project.minimum_renku_version); 
File \"/usr/lib/python3.10/site-packages/renku/infrastructure/database.py\", line 435, in setstate; 
self._reader.set_ghost_state(object, data); 
File \"/usr/lib/python3.10/site-packages/renku/infrastructure/database.py\", line 973, in set_ghost_state; 
state = self._deserialize_helper(data, create=False); 
File \"/usr/lib/python3.10/site-packages/renku/infrastructure/database.py\", line 1066, in _deserialize_helper; 
data = self._deserialize_helper(data); 
File \"/usr/lib/python3.10/site-packages/renku/infrastructure/database.py\", line 1012, in _deserialize_helper; 
data[key] = self._deserialize_helper(value); 
File \"/usr/lib/python3.10/site-packages/renku/infrastructure/database.py\", line 1027, in _deserialize_helper; 
cls = self._get_class(object_type); 
File \"/usr/lib/python3.10/site-packages/renku/infrastructure/database.py\", line 955, in _get_class; 
cls = get_class(type_name); 
File \"/usr/lib/python3.10/site-packages/renku/infrastructure/database.py\", line 112, in get_class; 
return get_attribute(module, components[1:]); 
File \"/usr/lib/python3.10/site-packages/renku/infrastructure/database.py\", line 145, in get_attribute; 
object = getattr(object, component); 
AttributeError: module 'renku.domain_model.project' has no attribute 'ProjectTemplateMetadata';
```
@Panaetius maybe you can give us some help, please.

Reply:
@jachro Ok! I just talked to Ralf, and he modified this CI deployment so I can make a test of WFF-lineage view. I will do that and report back the results!